compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/internal/ClipboardUtils.desktop.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.foundation.internal
+
+import androidx.compose.ui.platform.ClipEntry
+import androidx.compose.ui.platform.NativeClipboard
+import androidx.compose.ui.text.AnnotatedString
+import java.awt.datatransfer.DataFlavor
+import java.awt.datatransfer.StringSelection
+import java.io.IOException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal actual suspend fun ClipEntry.readText(): String? {
+    if (!hasText()) return null
+
+    return withContext(Dispatchers.IO) {
+        try {
+            transferable.getTransferData(DataFlavor.stringFlavor) as? String
```

Comment:
If it shouldn't throw an exceptions, I'd go with `runCatching { ... }.getOrNull()` here, to catch all exceptions.